src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -39,3 +48,75 @@ async def exec_aggregation_async(
     ) -> AggregationResult:
         """Execute aggregation nodes"""
         raise NotImplementedError()
+
+
+class APIBasedExecutorProxy(AbstractExecutorProxy):
+    @property
+    def api_endpoint(self) -> str:
+        """The basic API endpoint of the executor service.
+
+        The executor proxy calls the executor service to get the
+        line results and aggregation result through this endpoint.
+        """
+        raise NotImplementedError()
+
+    async def exec_line_async(
+        self,
+        inputs: Mapping[str, Any],
+        index: Optional[int] = None,
+        run_id: Optional[str] = None,
+    ) -> LineResult:
+        # ensure service health
+        await self.ensure_executor_health()
+        # call execution api to get line results
+        url = self.api_endpoint + "/Execution"
+        payload = {"run_id": run_id, "line_number": index, "inputs": inputs}
+        async with httpx.AsyncClient() as client:
+            response = await client.post(url, json=payload, timeout=LINE_TIMEOUT_SEC)
+        response = self.process_http_response(response)
+        return LineResult.deserialize(response)
+
+    async def exec_aggregation_async(
+        self,
+        batch_inputs: Mapping[str, Any],
+        aggregation_inputs: Mapping[str, Any],
+        run_id: Optional[str] = None,
+    ) -> AggregationResult:
+        # ensure service health
+        await self.ensure_executor_health()
+        # call aggregation api to get aggregation result
+        async with httpx.AsyncClient() as client:
+            url = self.api_endpoint + "/Aggregation"
+            payload = {"run_id": run_id, "batch_inputs": batch_inputs, "aggregation_inputs": aggregation_inputs}
+            response = await client.post(url, json=payload, timeout=LINE_TIMEOUT_SEC)
+        response = self.process_http_response(response)
+        return AggregationResult.deserialize(response)
+
+    def process_http_response(self, response: httpx.Response):
+        status_code = response.status_code
+        if status_code == 200:
+            return response.json()
+        else:
+            # TODO: add more error handling
+            raise PromptflowException(f"Error when calling executor API, response: {response}")
+
+    async def ensure_executor_health(self):
```

Comment:
add some docstring for this and add leave something about the observation that the service starts relatively slow in Linux

Reply:
Added.